iFixit/react-commerce
# ProductList: Add Optional Filters

## Overview

We wanted to be able to highlight some products on product list pages by surfacing or boosting them to the top of the results. This pull allows us to do that by using algolia's optionalFIlters.

## QA 

You'll want to find a productList to use. Go to it in strapi and add data to the optionalFilters field

For example I had something like for `/Parts/Google_Pixel_6_Pro`

```
category:Adhesive Strips
identifiers:356-324
identifiers:356-080
```

This will boost all products that match these conditions in algolia, and should boost the products that match more of these conditions more. So one of the items matches `356-080` `Adhesive Strips` so it should be at the top of the results.


If you want to read about option filter rules read https://www.algolia.com/doc/guides/managing-results/rules/merchandising-and-promoting/in-depth/optional-filters/

Note: you the strapi optionalFilters are split by endline characters, and in each line are split by `||` if you want to do a more complex filter. So

```
category:Adhesive Strips
identifiers:356-324 || identifiers:356-080
```
would result in the optional filters of

`['category:Adhesive Strips', ['identifiers:356-324', 'identifiers:356-080']]`

Closes: https://github.com/iFixit/ifixit/issues/50010
